src/lightning_app/cli/commands/connection.py

```diff
@@ -244,22 +258,37 @@ def _list_app_commands(echo: bool = True) -> List[str]:
         click.echo("The current Lightning App doesn't have commands.")
         return []
 
+    app_info = metadata[command_names[0]].get("app_info", None)
```

Comment:
what happens if there are no commands? Getting index `0` in an empty list would raise an unhandled exception.

Reply:
this is actually handled by the previous condition `if command_names`